internal/appsec/waf_test.go

```diff
@@ -198,6 +199,15 @@ func TestBlocking(t *testing.T) {
 		}
 		w.Write([]byte("Hello World!\n"))
 	})
+	mux.HandleFunc("/body", func(w http.ResponseWriter, r *http.Request) {
+		buf := new(strings.Builder)
+		io.Copy(buf, r.Body)
+		if err := pAppsec.MonitorParsedHTTPBody(r.Context(), buf.String()); err != nil {
+			w.Write([]byte(err.Error()))
```

Comment:
Our action handler actually writes the response in that case.
Doing this is fine but the content you write shouldn't be expected in the response, and the result will be the same as simply returning because the response is already committed.

See `NewBlockRequestAction`